imagetagger/imagetagger/images/templates/images/download.py

```diff
@@ -1,154 +1,219 @@
-    error = False
-    if not os.path.exists(os.path.join(os.getcwd(), filename, current_imageset)):
-        os.makedirs(os.path.join(os.getcwd(), filename, current_imageset))
-    page = requests.get("{}images/imagelist/{}/".format(BaseUrl,
-                        current_imageset),
-                        cookies=cookies)
+
+    print(f"Downloaded imageset {set_id}")
+    return True
+
+
+def download_imageset(set_id, target_folder, login_data):
+    """
+    Downloads an imageset as separate files
+
+    :param set_id: The id of the image set to download
+    :param target_folder: The folder in which the images should be saved
+    :param login_data: Data as returned by login()
+    """
+    print(f'Now downloading {set_id}')
+
+    os.makedirs(target_folder, exist_ok=True)
+
+    page = requests.get(f"{BASE_URL}images/imagelist/{set_id}/", cookies=login_data[1])
     if page.status_code == 404:
-        print("In Imageset {} was an error. The server returned page not found.".format(current_imageset))
-        errorlist.append(current_imageset)
-        return
+        print("In Imageset {} was an error. The server returned page not found.".format(set_id))
+        return False
+
     images = page.text.replace('\n', '')
     images = images.split(',')
+    error = True
+
     for index, image in enumerate(images):
         if image == '':
             continue
-        r = requests.get(BaseUrl + image[1:],
-                         data=data,
-                         cookies=cookies,
+        r = requests.get(BASE_URL + image[1:],
+                         data=login_data[0],
+                         cookies=login_data[1],
                          allow_redirects=False,
-                         headers={'referer': BaseUrl},
+                         headers={'referer': BASE_URL},
                          stream=True)
         if r.status_code == 404:
-            print("In Imageset {} was an error. The server returned page not found.".format(current_imageset))
-            errorlist.append(current_imageset)
-
+            print("In Imageset {} was an error. The server returned page not found.".format(set_id))
             error = True
             continue
+
         image = image.split('?')[1]
-        with open(os.path.join(filename, current_imageset, image), 'wb') as f:
+        with open(os.path.join(target_folder, image), 'wb') as f:
             r.raw.decode_content = True
             shutil.copyfileobj(r.raw, f)
             sys.stdout.flush()
-            print("{}Image {} / {} has been downloaded from imageset {}".format("\r", index + 1, len(images) - 1, current_imageset), end="")
+            print(f"\rImage {index + 1} / {len(images) - 1} has been downloaded from imageset {set_id}", end="")
+
+    print()
+
     if not error:
-        print('\nImageset {} has been downloaded.'.format(current_imageset))
+        print('\nImageset {} has been downloaded.'.format(set_id))
+    return error
```

Comment:
Am i overlooking something or are we always returning True here?
In line 99 we set error to True and i did not find a line where it can be set to False
This also means line 112 does nothing.

I assume we want to return False if something did not work, since that is how we handle it if the set throws a 404 (line 95)? 
If so we should probably rename error to something like no_error and change it's value in line 112